lib/practice_exercises.rb

```diff
@@ -1,13 +1,52 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity:  O(1)
+# Space Complexity: O(n)
+#divide the string in half, create a variable called the mid_point(interator)
+# set equal to true 
+# zero variable equals i 
+# while i is less than the mid_point keep going..
+# seeing if  string[i]is equal to that same i minus on
+#the aaray is now only looking at the index 3 
+#[string.length-i-1] will point it to anything before mid_point 
+# if when comparing the first have of the string and last they are not the same 
+#return false
+#however if true to will move to the next to compare
+#this is achieved by cutting the aaray in halfddd 
 def is_palindrome(string)
```

Comment:
How could the time complexity be constant?  It depends on the length of the string so this solution is O(n).  Also since you're not making any further strings or arrays, this solution has space complexity of O(1). 

You have most of this working, just check my comments for suggestions for working with space and punctuation.